include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -2952,13 +2952,15 @@ namespace internal
       set_dof_indices (DoFCellAccessor<DoFHandler<3,spacedim>, level_dof_access> &accessor,
                        const std::vector<types::global_dof_index>          &local_dof_indices)
       {
+        const unsigned int dim = 3;
+
         Assert (accessor.has_children() == false,
                 ExcInternalError());
 
-        const unsigned int dofs_per_vertex = accessor.get_fe().dofs_per_vertex,
-                           dofs_per_line   = accessor.get_fe().dofs_per_line,
-                           dofs_per_quad   = accessor.get_fe().dofs_per_quad,
-                           dofs_per_hex    = accessor.get_fe().dofs_per_hex,
+        const unsigned int dofs_per_vertex = accessor.get_fe().template n_dofs_per_object<0>(),
+                           dofs_per_line   = accessor.get_fe().template n_dofs_per_object<1>(),
+                           dofs_per_quad   = accessor.get_fe().template n_dofs_per_object<2>(),
+                           dofs_per_hex    = accessor.get_fe().template n_dofs_per_object<3>(),
```

Comment:
I find this new version harder to read than the one before. Are you going to take advantage of this new structrure in upcoming PRs or what is the reason for accessing these numbers by `n_dofs_per_object` instead of the name?

Reply:
Hm, yes, you're completely right. I don't know what I was thinking.

I have a bunch of other patches lined up after this one already that duplicate, move, and merge this function to other places. Would you be willing to merge this patch as is if I promise to make the reverse change in a follow-up patch? Otherwise, none of my later patches will apply any more if I undo this part of the commit :-(

Reply:
Fine with me.

Reply:
Appreciated!